### DISCUSSION TITLE
health based routing between multiple xfrm interfaces

### DISCUSSION THREAD

**Discussion by nirdothan** [strongswan/strongswan#1931] [Q&A] [answered]
Hi,
I'm trying to establish a site-to-site vpn where one gateway has multiple tunnels to the other side in an HA configuration.
According to my understanding, if I want to keep both tunnels up, they need be route based and so I followed the [documentation](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_xfrm_interfaces_on_linux). I have 2 connections and 2 xfrm interfaces associated with them.  The routing table is setup with duplicate entries where one has a lower metric than the other. 
I expected that once I terminate the higher priority (lower metric route) connection, the xfrm interface would detect the dead link and the lower priority route would be used thus routing to the other connection. 
This does not seem to happen and the system keeps trying to send traffic to the dead link. 
Please explain if I've misunderstood or if any special configuration is missing. 
Thanks

**Comment by tobiasbrunner** [comment]
> if I want to keep both tunnels up, they need be route based

Not necessarily.

> I expected that once I terminate the higher priority (lower metric route) connection, the xfrm interface would detect the dead link

Why would the XFRM interface detect anything? If you are referring to the "underlying interface", that's basically ignored by the kernel (the only effect is has is that the MAC address is copied, that's why configuring one is optional for newer kernels and tools). If you are referring to not having any SAs or policies associated with the interface (once the connection is down), that won't have an effect on the interface's state either.  So those routes won't disappear automatically.

**Reply by nirdothan** [reply]
Thanks @tobiasbrunner
My understanding was based on a configuration file that I downloaded from AWS for a StrongSwan customer gateway. The file begins with the following text.

> this configuration consists of two tunnels. Both tunnels must be configured on your Customer Gateway. If you are configuring your tunnels as policy-based, only a single tunnel may be up at a time. If you are configuring your tunnels as route-based, both tunnels may be up simultaneously

Further down it instructs to create 2 VTI interfaces with routes etc. I can send the whole file if it helps.
Since my understanding was incorrect, can you please advise how to implement an active-active dual tunnel solution. It can be policy based or route based. Whatever works best.

**Reply by tobiasbrunner** [reply]
Hm, why would that make a difference to the responder? Unless the config is actually different for the policy vs. route approach (e.g. the traffic selectors).

"Whatever works best" probably depends on a lot of things. But if you just want the routes gone if the SA is gone, install them dynamically in an updown or vici script or let strongSwan manage routes via XFRM interface (available in newer strongSwan versions).

**Reply by nirdothan** [reply, marked as answer]
Thanks @tobiasbrunner I now understand what I was doing wrong
